documentation/src/main/asciidoc/topics/rest_api_v2.adoc

```diff
@@ -429,27 +486,104 @@ Request responses depend on the type of counter, as follows:
 This method processes only `plain/text` content.
 ====
 
-[[rest_server_counters_get]]
-==== Getting Counter Values
 To retrieve the value of a counter, invoke a `GET` request.
 
 [source,options="nowrap",subs=attributes+]
 ----
 GET /v2/counters/{counterName}
 ----
 
-[NOTE]
-====
-This method processes only `plain/text` content.
-====
+.Headers
+|===
+|Header |Required or Optional |Parameter
 
-[[rest_server_counters_delete]]
-==== Resetting Counters
-To reset counters to their initial values, invoke a `DELETE` request.
+|`link:#rest_accept[Accept]`
+|OPTIONAL
+|The required format to return the content. Supported formats are _application/json_ and _text/plain_. JSON is assumed if no header is provided.
+|===
+
+
+[[rest_server_counters_reset]]
+=== Reseting Counters
+
+To reset counters, use a `GET` request with the `?action=reset` parameter.
 
 [source,options="nowrap",subs=attributes+]
 ----
-DELETE /v2/counters/{counterName}
+GET /v2/counters/{counterName}?action=reset
 ----
 
-//-
+[[rest_server_counters_inc_weak]]
+=== Incrementing Weak Counters
+
+To increment a Weak Counter, use the `?action=increment`parameter.
+
+[source,options="nowrap",subs=attributes+]
+----
+GET /v2/counters/{counterName}?action=increment
+----
+
+[[rest_server_counters_add_weak]]
+=== Adding a delta to Weak Counters
+
+To add an arbitrary amount to a Weak Counter, use the params `?action=add` and `delta`.
+
+[source,options="nowrap",subs=attributes+]
+----
+GET /v2/counters/{counterName}?action=add&delta={delta}
+----
+
+[[rest_server_counters_inc_strong]]
+=== IncrementAndGet Strong Counters
+
+[source,options="nowrap",subs=attributes+]
+----
+GET /v2/counters/{counterName}?action=incrementAndGet
```

Comment:
can it use the same action name as the weak counter? 
Only the return value changes with the counter's type (nothing for weak, the new value for strong)

Reply:
Not sure...If we call it ```incrementAndGet``` for both, doesn't it imply a CAS operation for strong and also weak?

Reply:
OTOH, calling it ```increment```, would it miss the fact that for strong it's also a get ?

Reply:
For Hot Rod, there is a single "increment" operation but the invocation depends in counter's type. 
https://github.com/infinispan/infinispan/blob/master/server/hotrod/src/main/java/org/infinispan/server/hotrod/CounterRequestProcessor.java#L122

For CAS, I'm throwing an exception if it is a weak counter:
https://github.com/infinispan/infinispan/blob/master/server/hotrod/src/main/java/org/infinispan/server/hotrod/CounterRequestProcessor.java#L91

Reply:
I think having a single increment is reasonable, will change it

Reply:
So just to be clear, we'd have only one ```increment```, only one ```decrement``` and only one ```add```, treating them differently (return type) depending on the weak/strong case? Is this how Hot Rod is organized?

Reply:
yes